components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/java/org/wso2/carbon/ndatasource/ui/NDataSourceHelper.java

```diff
@@ -498,22 +501,27 @@ private static void setDatasourceProperties(RDBMSDSXMLConfiguration rdbmsDSXMLCo
                         }
                 }
 	}
-	
-	public static String elementToString(Element element) {
-		try {
-			if (element == null) {
-				return null;
-			}
-		    Transformer transformer = TransformerFactory.newInstance().newTransformer();
-		    StringWriter buff = new StringWriter();
-		    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-		    transformer.transform(new DOMSource(element), new StreamResult(buff));
-		    return buff.toString();
-		} catch (Exception e) {
-			log.error("Error while convering element to string: " + e.getMessage(), e);
-			return null;
-		}
-	}
+
+    public static String elementToString(Element element) {
+        try (StringWriter buff = new StringWriter()) {
+            if (element == null) {
+                return null;
+            }
+            TransformerFactory factory = TransformerFactory.newInstance();
+            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            Transformer transformer = factory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            transformer.transform(new DOMSource(element), new StreamResult(buff));
+            return buff.toString();
+        } catch (TransformerException e) {
+            log.error("Error while converting element to string: " + e.getMessage(), e);
+            return null;
```

Comment:
There is a chance of getting a null pointer exception in [1]  since are returning null here and in line 522.Shall we handle it properly?  

[1] - https://github.com/nalakas/carbon-commons-1/blob/a2866804234525e60278878ad1fd693af1fa762f/components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/java/org/wso2/carbon/ndatasource/ui/NDataSourceHelper.java#L581

Reply:
fixed in 488d543